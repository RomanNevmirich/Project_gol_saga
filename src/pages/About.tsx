import React from 'react';
import Header from '../components/Header/Header';

const About = () => (
  <>
    <Header />
    <h1>Почему не видно текста????</h1>
  </>
);

export default About;