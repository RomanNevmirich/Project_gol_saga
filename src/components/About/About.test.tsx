import React from 'react';
import { render } from '@testing-library/react';
import { About } from './About';
import { HashRouter } from "react-router-dom";

describe('About component check', () => {
  
  it('About render', () => {
    const { asFragment } = render(<HashRouter><About /></HashRouter>);

    expect(asFragment()).toMatchSnapshot();
  });

});