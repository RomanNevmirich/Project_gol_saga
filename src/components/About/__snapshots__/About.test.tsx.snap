// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component check About render 1`] = `
<DocumentFragment>
  <nav>
    <a
      href="#/"
    >
      Вернуться на главную
    </a>
  </nav>
  <main>
    <h3>
      Правила
    </h3>
    <p>
      Место действия игры — размеченная на клетки плоскость, которая может быть безграничной, ограниченной, или замкнутой.
    </p>
    <p>
      Каждая клетка на этой поверхности имеет восемь соседей, окружающих её, и может находиться в двух состояниях: быть «живой» (заполненной) или «мёртвой» (пустой).
    </p>
    <p>
      Распределение живых клеток в начале игры называется первым поколением. Каждое следующее поколение рассчитывается на основе предыдущего по таким правилам: в пустой (мёртвой) клетке, с которой соседствуют три живые клетки, зарождается жизнь; если у живой клетки есть две или три живые соседки, то эта клетка продолжает жить; в противном случае (если живых соседей меньше двух или больше трёх) клетка умирает («от одиночества» или «от перенаселённости»).
    </p>
    <p>
      Игра прекращается, если на поле не останется ни одной «живой» клетки; конфигурация на очередном шаге в точности (без сдвигов и поворотов) повторит себя же на одном из более ранних шагов (складывается периодическая конфигурация) при очередном шаге ни одна из клеток не меняет своего состояния (предыдущее правило действует на один шаг назад, складывается стабильная конфигурация)
    </p>
    <p>
      Игрок не принимает активного участия в игре. Он лишь расставляет или генерирует начальную конфигурацию «живых» клеток, которые затем изменяются согласно правилам. Несмотря на простоту правил, в игре может возникать огромное разнообразие форм.
    </p>
    <a
      href="https://ru.wikipedia.org/wiki/%D0%98%D0%B3%D1%80%D0%B0_%C2%AB%D0%96%D0%B8%D0%B7%D0%BD%D1%8C%C2%BB"
      target="_blank"
    >
      Описание на Wiki
    </a>
    <p />
  </main>
</DocumentFragment>
`;
